crates/diagnostic_tests/src/zomes.rs

```diff
@@ -0,0 +1,146 @@
+                    ))?;
+                    let _ = api.create_link(CreateLinkInput::new(
+                        base.clone(),
+                        hash.clone().into(),
+                        ZomeIndex(0),
+                        LinkType::new(0),
+                        ().into(),
+                        ChainTopOrdering::Relaxed,
+                    ))?;
+                }
+                Ok(())
+            },
+        )
+        .function(
+            "link_count",
+            |api, (base, entries): (AnyLinkableHash, bool)| {
+                let links = api
+                    .get_links(vec![GetLinksInput::new(
+                        base,
+                        LinkTypeFilter::single_dep(0.into()),
+                        None,
+                    )])
+                    .unwrap();
+                let links = links.first().unwrap();
+                if entries {
+                    let gets = links
+                        .iter()
+                        .map(|l| {
+                            let target = l.target.clone().retype(holo_hash::hash_type::Action);
+                            GetInput::new(target.into(), Default::default())
+                        })
+                        .collect();
+                    let somes = api
+                        .get(gets)
+                        .unwrap()
+                        .into_iter()
+                        .filter(|e| e.is_some())
+                        .count();
+                    Ok(somes)
+                } else {
+                    Ok(links.len())
+                }
+            },
+        )
+        .function("validate", |_api, _op: Op| {
+            Ok(ValidateCallbackResult::Valid)
+        })
+}
+
+pub fn syn_zome() -> InlineIntegrityZome {
+    InlineIntegrityZome::new_unique([EntryDef::from_id("a")], 0)
+        .function("commit", |api, bytes: Vec<u8>| {
+            let entry: SerializedBytes = UnsafeBytes::from(bytes).try_into().unwrap();
+            api.create(CreateInput::new(
+                InlineZomeSet::get_entry_location(&api, EntryDefIndex(0)),
+                EntryVisibility::Public,
+                Entry::App(AppEntryBytes(entry)),
+                ChainTopOrdering::Relaxed,
+            ))?;
+            Ok(())
+        })
+        //
+        .function(
+            "send_message",
+            |api, (msg, agents): (Vec<u8>, Vec<AgentPubKey>)| {
+                api.remote_signal(RemoteSignal {
+                    agents,
+                    signal: ExternIO::encode(msg).unwrap(),
+                })?;
+
+                // let res = api.call(vec![Call {
+                //     target: CallTarget::NetworkAgent(agents[1].clone()),
+                //     zome_name: "zome".into(),
+                //     fn_name: "recv_remote_signal".into(),
+                //     cap_secret: None,
+                //     payload: msg.into(),
+                // }])?;
+                // dbg!(res);
```

Comment:
There's a lot of commented code throughout this PR

Reply:
Sorry I need to get better about cleaning these things up